### PR TITLE
Implement getFirebaseProject and deprecate firebaseApi

### PR DESCRIPTION
**Pull request by TrCaM**
### Description

- Implement `getFirebaseProject` in `projects.ts` that fetch information of a Firebase project given `projectId` as input
- Deprecate `firebaseApi`, and replace code that use `firebaseApi` with API methods from `management/projects.ts`
- Update unit tests

### Testing
- Manual testing with `firebase init` and `firebase use` commands, which were users of `firebaseApi`

### Next step
- Update choosing projects from a list behavior:
     + List a single page of maximum of 100 projects
     + If next page token is provided then prompt user for `projectId` string
- Update [create a new project] option while choosing a project of `firebase init` to actually create a new Firebase project and choose it as the working project for the directory. 
- Deprecate `firebase list` command

### CODE REVIEW

File: src/management/projects.ts [hunk 1]
```diff
@@ -12,14 +12,14 @@ export interface FirebaseProjectMetadata {
   projectId: string;
   projectNumber: string;
   displayName: string;
-  resources: DefaultProjectResources;
+  resources?: DefaultProjectResources;
 }
 
 export interface DefaultProjectResources {
```
Comment (bkendall) [inline src/management/projects.ts]: Are _all_ these properties really optional now? Why that change?

Comment (TrCaM) [reply src/management/projects.ts]: it turns out that these resources may not exist in a Firebase project, for example with deferred location a new project can be created without specifying `locationId`.

Comment (bkendall) [reply src/management/projects.ts]: huh. the more you know.

Comment (thechenky) [reply src/management/projects.ts]: Confirming this is true. Due to deferred project resource creation, all should be optional now.

**Pull request merged**